Experimental/CameraComponent/Viewport.cs

```diff
@@ -0,0 +1,322 @@
+                        "0, Viewport.Size.X, 0, 0, " +
+                        "0, 0, Viewport.Size.X, 0, " +
+                        "0, 0, 0, 1)";
+                    break;
+                case 2f: // FixY
+                    stretchMat = "Matrix4x4(" +
+                        "Viewport.Size.Y, 0, 0, 0, " +
+                        "0, Viewport.Size.Y, 0, 0, " +
+                        "0, 0, Viewport.Size.Y, 0, " +
+                        "0, 0, 0, 1)";
+                    break; 
+                case 3f:
+                    Vector2 uniform_size;
+                    propertySet.TryGetVector2("Size", out uniform_size);
+
+                    if (uniform_size.X >= uniform_size.Y)
+                    {
+                        stretchMat = "Matrix4x4(" +
+                        "Viewport.Size.Y, 0, 0, 0, " +
+                        "0, Viewport.Size.Y, 0, 0, " +
+                        "0, 0, Viewport.Size.Y, 0, " +
+                        "0, 0, 0, 1)";
+                    }
+                    else
+                    {
+                        stretchMat = "Matrix4x4(" +
+                        "Viewport.Size.X, 0, 0, 0, " +
+                        "0, Viewport.Size.X, 0, 0, " +
+                        "0, 0, Viewport.Size.X, 0, " +
+                        "0, 0, 0, 1)";
+                    }
+                    break;
+                case 4f:
+                    Vector2 uniform_to_fill_size;
+                    propertySet.TryGetVector2("Size", out uniform_to_fill_size);
+
+                    if (size.X >= size.Y)
+                    {
+                        stretchMat = "Matrix4x4(" +
+                        "Viewport.Size.X, 0, 0, 0, " +
+                        "0, Viewport.Size.X, 0, 0, " +
+                        "0, 0, Viewport.Size.X, 0, " +
+                        "0, 0, 0, 1)";
+                    }
+                    else
+                    {
+                        stretchMat = "Matrix4x4(" +
+                        "Viewport.Size.Y, 0, 0, 0, " +
+                        "0, Viewport.Size.Y, 0, 0, " +
+                        "0, 0, Viewport.Size.Y, 0, " +
+                        "0, 0, 0, 1)";
+                    }
+                    break;
+                default:
+                    stretchMat = "";
+                    break;
+            }
+            
+            var stretchExpression = Window.Current.Compositor.CreateExpressionAnimation();
+            stretchExpression.Expression = stretchMat;
+            stretchExpression.SetReferenceParameter("Viewport", propertySet);
+
+            return stretchExpression;
+        }
+
+        private void Camera_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(!Camera.UseAnimations)
+            {
+                this.CreateCameraMatrix();
+            }
+            else
+            {
+                this.CreateExpressionAnimation();

```

Comment:
I suspect this is definitely overkill.  CreateExpressionAnimation inserts values into the propertySet, effectively obliterating all currently running animations.  Since you are sending information about which things have changed, I think you should be able to just ignore a bunch of them, and hopefully only insert a single new value into a PropertySet and not have to regenerate any expression animations.